File: app/src/main/java/com/monk/retrofit/mvp/view/RetrofitView.java
package com.monk.retrofit.mvp.view;

import com.monk.retrofit.base.IBaseView;
import com.monk.retrofit.bean.Bean;

/**
 * @author monk
 * @date 2019-02-19
 */
public interface RetrofitView extends IBaseView {
    void showProgressBar();

    void dismissProgressBar();

    void showError(String error);

    void displayBean(Bean bean);
}

File: module_fragment/src/main/java/com/monk/modulefragment/retrofit/mvp/view/RetrofitView.kt
package com.monk.modulefragment.retrofit.mvp.view

import com.monk.modulefragment.retrofit.base.IBaseView
import com.monk.modulefragment.retrofit.bean.Bean

/**
 * @author monk
 * @date 2019-02-19
 */
interface RetrofitView : IBaseView {
    fun showProgressBar()
    fun dismissProgressBar()
    fun showError(error: String?)
    fun displayBean(bean: Bean?)
}